src/utils/subgraph.ts

```diff
@@ -151,53 +220,68 @@ export async function getPrice(asset: DDO): Promise<BestPrice> {
     datatokenAddress: asset?.dataToken.toLowerCase()
   }
 
-  const poolPriceResponse: any = await fetchData(
+  const poolPriceResponse: ApolloQueryResult<AssetsPoolPrice> = await fetchData(
     AssetPoolPriceQuerry,
     poolVariables
   )
-  const frePriceResponse: any = await fetchData(AssetFreQuery, freVariables)
-  if (poolPriceResponse.data?.pools.length > 0) {
-    const price: BestPrice = {
-      type: 'pool',
-      address: poolPriceResponse.data?.pools[0]?.id,
-      value:
-        poolPriceResponse.data?.pools[0]?.consumePrice === '-1'
-          ? poolPriceResponse.data?.pools[0]?.spotPrice
-          : poolPriceResponse.data?.pools[0]?.consumePrice,
-      ocean: poolPriceResponse.data?.pools[0]?.oceanReserve,
-      datatoken: poolPriceResponse.data?.pools[0]?.datatokenReserve,
-      pools: [poolPriceResponse.data?.pools[0]?.id],
-      isConsumable:
-        poolPriceResponse.data?.pools[0]?.consumePrice === '-1'
-          ? 'false'
-          : 'true'
-    }
-    return price
-  } else if (frePriceResponse.data?.fixedRateExchanges.length > 0) {
-    // TODO Hacky hack, temporary™: set isConsumable to true for fre assets.
-    // isConsumable: 'true'
-    const price: BestPrice = {
-      type: 'exchange',
-      value: frePriceResponse.data?.fixedRateExchanges[0]?.rate,
-      address: frePriceResponse.data?.fixedRateExchanges[0]?.id,
-      exchange_id: frePriceResponse.data?.fixedRateExchanges[0]?.id,
-      ocean: 0,
-      datatoken: 0,
-      pools: [],
-      isConsumable: 'true'
-    }
-    return price
-  } else {
-    const price: BestPrice = {
-      type: '',
-      value: 0,
-      address: '',
-      exchange_id: '',
-      ocean: 0,
-      datatoken: 0,
-      pools: [],
-      isConsumable: 'false'
-    }
-    return price
+  const frePriceResponse: ApolloQueryResult<AssetsFrePrice> = await fetchData(
+    AssetFreQuery,
+    freVariables
+  )
+
+  const bestPrice: BestPrice = transformPriceToBestPrice(
+    frePriceResponse.data.fixedRateExchanges,
+    poolPriceResponse.data.pools
+  )
+
+  return bestPrice
+}
+
+export async function getAssetsPrices(
```

Comment:
Some refactor on `getAssetPrices` would be nice and get rid of code duplications. Instead of having 2 methods that do the same thing but return different values you can have a base method that returns the graph `pool` and `fre prices`, and other 2 methods  responsible for our data processing, one returning `Promise<PriceList>` and the other one `Promise<AssetListPrices[]>.`
I would keep in the base method the following block https://github.com/oceanprotocol/market/blob/feature/use_asset_price_in_teaser/src/utils/subgraph.ts#L244-L265 return the values and move everything else out in those two new methods